ui-components/urldialog.md

```diff
@@ -7,91 +7,90 @@ keywords: dialog urldialog api
 
-**Name:** The name property on the button is used as an ID attribute to identify the dialog component. For example, when `name: foobutton` is defined and a user clicks on that button, the dialog `onAction()` handler will fire and provide an object containing the name of the dialog component, e.g., `details.name = 'foobutton'`. This will allow developers to create a click handler for **foobutton**.
+#### Configuration
 
-**Text:** This will be the text displayed on the button. For example, `text: ‘do magic’` will create a button with text **do magic**.
+| Name | Type | Requirement | Description |
+| ---- | ---- | ----------- | ----------- |
+| type | `'button'` | required | The component type. Must be `'button'`. |
+| text | string | required | Text to display in the button **if `icon` is not specified**. Also used for the button's `title` attribute. |
+| name | string | optional | A identifier for the button. If not specified, the button will be assigned a randomly generated `name`.  |
+| icon | string | optional | Name of the icon to be displayed. Must correspond to an icon in the icon pack. **When configured, the button will display the icon instead of text.** |
+| primary | boolean | optional | Whether to style the button as a primary or secondary button. |
+| disabled | boolean | optional | default: `false` - When `true`, the button will be disabled when the dialog loads. |
+| align | `'end'` or `'start'` | optional | default: `'end'` - When set to `'end'` the button will display on the right-hand side of the dialog. When set to `'start'` the button will display on the left-hand side. |
 
-**Icon:** This will be the name of the icon to be displayed on the button, in place of any text. The name must correspond to an icon in the icon pack. Dialog buttons do not support mixing icons and text at the moment.
+> Note: Buttons do not support mixing icons and text at the moment.
 
-**Disabled:** (Value: Boolean; Default: `False`): When set to `true`, the button will be disabled when the dialog loads.
+#### Button types and event callbacks
 
-**Primary:** (Default: `False`): When set to `true`, the button will be colored to stand out. The color will depend on the chosen [skin]({{site.baseurl}}/general-configuration-guide/customize-ui/#skins).
+The different footer button types will invoke different callbacks when clicked:
 
-**Align:** (Default: 'end'): This will define the position of the button in the footer. When set to `end`, the button will be positioned on the right side of the dialog. When set to `start`, the button will be positioned on the left side of the dialog.
+* A **Cancel** type button will invoke the `onCancel` and `onClose` callback functions provided in the dialog configuration. These callback functions are also fired when a user clicks the `X` button in the top right of the dialog. **TinyMCE assumes that a dialog will not have multiple Cancel buttons.**
```

Comment:
`TinyMCE assumes that a dialog will not have multiple Cancel buttons.` is confusing and misleading to me. I'd suggest just removing it or rewording it since two buttons are possible, it's just that they'd always have the same effect (so it wouldn't make much sense to include two).